presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java

```diff
@@ -587,7 +591,28 @@ public HiveSessionProperties(HiveClientConfig hiveClientConfig, OrcFileWriterCon
                         ENABLE_LOOSE_MEMORY_BASED_ACCOUNTING,
                         "Enable loose memory accounting to avoid OOMing existing queries",
                         hiveClientConfig.isLooseMemoryAccountingEnabled(),
-                        false));
+                        false),
+                booleanProperty(
```

Comment:
Since this feature can be controlled per query level, the node split stats (getTotalSplitsWeight / getQueuedSplitsWeightForStage) would actually not be accurate

For example, the feature is turned off by default, and only one query has this feature turned on. For this query, the split stats from previous queries (which are still running) are actually using different split weight calculation standards than this query

Reply:
In an environment with some queries enabling this feature and other queries not enabling it:

- It's true that different weight calculations would be used for the splits associated with the queries (disabled: all splits are the "standard" weight, enabled: each splits weight is calculated based on size in bytes)
- I wouldn't say that this means that the node split stats are not accurate, at all times the values are computed as the sum of the weight of the splits assigned assigned- it's just that the weights have been computed differently between queries. It's worth noting that the `getQueuedSplitsWeightForStage()` is a per-task metric, so there will be no interaction between queries on that metric, but the node level `getTotalSplitsWeight()` will have an interaction between the two queries. In a sense, this is unavoidable so long as this setting exists- but I don't think it's actually a problem since the current implementation when enabled will only weight splits as "smaller than standard" and never "larger than standard". The net effect is that queries with small files will still be allowed a higher level of parallelism than before, even when other queries with small files have this feature disabled.